src/config-import.js

```diff
@@ -0,0 +1,271 @@
+import yauzl from 'yauzl-promise'
+import { validate, valueSchemas } from '@mapeo/schema'
+import { json, buffer } from 'node:stream/consumers'
+import path from 'node:path'
+
+// Throw error if a zipfile contains more than 10,000 entries
+const MAX_ENTRIES = 10_000
+
+/**
+ * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
+ * `@types/yauzl-promise` has the wrong name for this key. We should
+ * eventually submit this patch upstream.
+ */
+/**
+ * @typedef {{
+ *   presets: { [id: string]: unknown }
+ *   fields:  { [id: string]: unknown }
+ * }} PresetsFile
+ */
+/**
+ * @typedef {Parameters<import('./icon-api.js').IconApi['create']>[0]} IconData
+ */
+
+/**
+ * @param {string} configPath
+ */
+export async function readConfig(configPath) {
+  /** @type {Error[]} */
+  const warnings = []
+
+  const zip = await yauzl.open(configPath)
+  if (zip.entryCount > MAX_ENTRIES) {
+    throw new Error(`Zip file contains too many entries. Max is ${MAX_ENTRIES}`)
+  }
+  /** @type {undefined | Entry} */
+  let presetsEntry
+  for await (const entry of zip) {
+    if (entry.filename === 'presets.json') {
+      presetsEntry = entry
+      break
+    }
+  }
+  if (!presetsEntry) {
+    throw new Error('Zip file does not contain presets.json')
+  }
+  const presetsFile = await json(await presetsEntry.openReadStream())
+  validatePresetsFile(presetsFile)
+
+  return {
+    get errors() {
```

Comment:
Yeah, renaming to warnings makes sense, since this should throw. 
The idea is to be able to grab warnings from the front-end to present them to the user in a friendly manner, but haven't sync with the frontend team in order to see how we would do this - maybe it can be addressed in a future PR? ? -
I think for now returning the warnings from importConfig makes sense